src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -959,6 +959,44 @@ void C2_MacroAssembler::neon_compare(FloatRegister dst, BasicType bt, FloatRegis
   }
 }
 
+void C2_MacroAssembler::neon_compare_zero(FloatRegister dst, BasicType bt, FloatRegister src,
+                                          int cond, bool isQ) {
+  SIMD_Arrangement size = esize2arrangement((unsigned)type2aelembytes(bt), isQ);
+  if (bt == T_FLOAT || bt == T_DOUBLE) {
+    switch (cond) {
+      case BoolTest::eq: fcm(Assembler::EQ, dst, size, src); break;
+      case BoolTest::ne: {
+        fcm(Assembler::EQ, dst, size, src);
+        notr(dst, isQ ? T16B : T8B, dst);
+        break;
+      }
+      case BoolTest::ge: fcm(Assembler::GE, dst, size, src); break;
+      case BoolTest::gt: fcm(Assembler::GT, dst, size, src); break;
+      case BoolTest::le: fcm(Assembler::LE, dst, size, src); break;
+      case BoolTest::lt: fcm(Assembler::LT, dst, size, src); break;
```

Comment:
The key to this problem of endless switch statements is a function from `BoolTest` cond to `Assembler::Condition`.

Such a function is `cmpOpOper(BoolTest::overflow).ccode()` .

Please use it everywhere a `BoolTest` needs to be converted to a `Condition`.

Reply:
Thanks for pointing this usage, I didn't know that before. I found that cmpOpOper is generated by cmpOp operand defined in aarch64.ad, and it is able to convert input BoolTest condition and Assembler condition. However, the vector compare operand in VectorMaskCmp is ConINode, while cmpOp operand can only match BoolNode. How about adding a new standalone function like x86(https://github.com/openjdk/jdk/blob/e245620f6f6a836aef8ddef9f699cc540f2a5eb6/src/hotspot/cpu/x86/x86.ad#L2498) to convert from BoolTest to Assembler::Condition?

Reply:
Maybe we can use cmpOpOper by following way in aarch64_vector.ad:

```
Assembler::Condition condition = (Assembler::Condition)(cmpOpOper((BoolTest::mask) (int)($cond$$constant)).ccode());
```

but I think this code style is a little ugly and cmpOpOper should be used as a operand but not a utility.

Reply:
For clarity, I agree that where there's existing code to do this in x86 we should copy its structure.